x/pairing/keeper/unresponsive_provider.go

```diff
@@ -29,14 +29,14 @@ func (k Keeper) UnstakeUnresponsiveProviders(ctx sdk.Context, epochsNumToCheckCU
 		largerEpochsNumConst = epochsNumToCheckCUForUnresponsiveProvider
 	}
 
-	// To check for punishment, we have to go back recommendedEpochNumToCollectPayment+max(epochsNumToCheckCUForComplainers,epochsNumToCheckCUForUnresponsiveProvider) epochs from the current epoch. If there isn't enough memory, do nothing
-	epochTemp := currentEpoch
-	for counter := uint64(0); counter < largerEpochsNumConst+recommendedEpochNumToCollectPayment; counter++ {
-		previousEpoch, err := k.epochStorageKeeper.GetPreviousEpochStartForBlock(ctx, epochTemp)
-		if err != nil {
-			return nil
-		}
-		epochTemp = previousEpoch
+	// To check for punishment, we have to go back:
+	//   recommendedEpochNumToCollectPayment+
+	//   max(epochsNumToCheckCUForComplainers,epochsNumToCheckCUForUnresponsiveProvider)
+	// epochs from the current epoch.
+	minHistoryBlock := k.getBlockEpochsAgo(ctx, currentEpoch, largerEpochsNumConst+recommendedEpochNumToCollectPayment)
+	if minHistoryBlock == 0 {
+		// not enough history, do nothing
+		return nil
 	}
```

Comment:
@orenl-lava are these changes suppose to be part of this PR?

Reply:
ugh. good catch.